lightning/src/blinded_path/message.rs

```diff
@@ -138,17 +136,25 @@ pub(super) fn blinded_hops<T: secp256k1::Signing + secp256k1::Verification>(
 ) -> Result<Vec<BlindedHop>, secp256k1::Error> {
 	let pks = intermediate_nodes.iter().map(|node| &node.node_id)
 		.chain(core::iter::once(&recipient_node_id));
-	let tlvs = pks.clone()
+	let tlvs: Vec<ControlTlvs> = pks.clone()
 		.skip(1) // The first node's TLVs contains the next node's pubkey
 		.zip(intermediate_nodes.iter().map(|node| node.short_channel_id))
 		.map(|(pubkey, scid)| match scid {
 			Some(scid) => NextMessageHop::ShortChannelId(scid),
 			None => NextMessageHop::NodeId(*pubkey),
 		})
 		.map(|next_hop| ControlTlvs::Forward(ForwardTlvs { next_hop, next_blinding_override: None }))
-		.chain(core::iter::once(ControlTlvs::Receive(ReceiveTlvs{ context: Some(context) })));
+		.chain(core::iter::once(ControlTlvs::Receive(ReceiveTlvs{ context: Some(context) })))
+		.collect();
```

Comment:
Do we need to `collect` here? I think we can just clone the iterator to use it in the `max_length` calculation.

Reply:
Discussed a bit offline, but I just wanted to revisit this comment. Cloning should be cheap and the least expensive after all since the iterator doesn't own the underlying data. When we chatted about this, I mistakenly questioned my initial comment and thought the iterator owned the data in this instance.

So remove the `clone` here and replace the `tlvs.iter()` below with `tlvs.clone()`. However, this does require deriving `Clone` for `ControlTlvs` and containing structs since `core::iter::Once` only implements `Clone` if the underlying data implements `Clone` because it *does* own the data.

Reply:
Thanks a lot for the detailed suggestion! 🚀
I have updated the code to use the approach you described [pr3177.02](https://github.com/shaavan/rust-lightning/commits/pr3177.02)